backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java

```diff
@@ -12,54 +12,157 @@
-                "Acknowledgements",
-                "More Information"};
-        XWPFTable table = createTableAndAddReleasesTableHeaders(document, tableHeaders);
-        fillReleasesTable(table, projectLicenseInfoResults);
+        for (LicenseInfoParsingResult parsingResult : projectLicenseInfoResults) {
+            addReleaseTitle(document, parsingResult);
+            if (parsingResult.getStatus() == LicenseInfoRequestStatus.SUCCESS) {
+                XWPFRun copyrightTitleRun = document.createParagraph().createRun();
+                addFormattedText(copyrightTitleRun, "Copyrights", FONT_SIZE, true);
+                for (String copyright : getReleaseCopyrights(parsingResult)) {
+                    XWPFParagraph copyPara = document.createParagraph();
+                    copyPara.setSpacingAfter(0);
+                    setText(copyPara.createRun(), copyright);
+                }
+                XWPFRun licensesTitleRun = document.createParagraph().createRun();
+                addNewLines(licensesTitleRun, 1);
+                addFormattedText(licensesTitleRun, "Licenses", FONT_SIZE, true);
+                for (String licenseName : getReleasesLicenses(parsingResult)) {
+                    XWPFParagraph licensePara = document.createParagraph();
+                    licensePara.setSpacingAfter(0);
+                    addHyperLink(licensePara, "bookmark_" + licenseName, licenseName);
+                }
+            } else {
+                XWPFRun errorRun = document.createParagraph().createRun();
+                String errorText = nullToEmptyString(parsingResult.getMessage());
+                String filename = getFilename(parsingResult);
+                addFormattedText(errorRun, String.format("Error reading license information: %s", errorText), FONT_SIZE, false, ALERT_COLOR);
+                addFormattedText(errorRun, String.format("Source file: %s", filename), FONT_SIZE, false, ALERT_COLOR);
+            }
+            addNewLines(document.createParagraph().createRun(), 1);
+        }
+        addPageBreak(document.createParagraph().createRun());
+    }
 
-        addLicenseTextsHeader(document, "Appendix - License Texts");
-        addLicenseTexts(document, projectLicenseInfoResults);
+    private void addReleaseTitle(XWPFDocument document, LicenseInfoParsingResult parsingResult) {
+        String releaseTitle = nullToEmptyString(parsingResult.getName()) + " " + nullToEmptyString(parsingResult.getVersion());
+        XWPFParagraph releaseTitleParagraph = document.createParagraph();
+        releaseTitleParagraph.setStyle(STYLE_HEADING);
+        addBookmark(releaseTitleParagraph, releaseTitle, releaseTitle);
+        addNewLines(document.createParagraph().createRun(), 0);
     }
-}
 
+    private Set<String> getReleaseCopyrights(LicenseInfoParsingResult licenseInfoParsingResult) {
+        Set<String> copyrights = Collections.emptySet();
+        if (licenseInfoParsingResult.isSetLicenseInfo()) {
```

Comment:
Why is this approach here using if statements compared to the approach in line 147ff using `Optional.ofNullable(..)`?